### ISSUE TITLE
FATAL ERROR: HandleScope::HandleScope Entering the V8 API without proper locking in place

### ISSUE THREAD

**Issue by BryceMehring** [DataDog/dd-trace-js#3379]
**Expected behaviour**

Node should not throw error.

**Actual behaviour**

FATAL ERROR being thrown from node when appsec mode is enabled.
```
FATAL ERROR: HandleScope::HandleScope Entering the V8 API without proper locking in place
 1: 0x10f417665 node::Abort() (.cold.1) [/.nvm/versions/node/v16.20.0/bin/node]
 2: 0x10e1101c9 node::Abort() [/.nvm/versions/node/v16.20.0/bin/node]
 3: 0x10e1102ba node::OnFatalError(char const*, char const*) [/.nvm/versions/node/v16.20.0/bin/node]
 4: 0x10e2874c2 v8::EscapableHandleScope::EscapableHandleScope(v8::Isolate*) [/.nvm/versions/node/v16.20.0/bin/node]
 5: 0x10e0c9c58 napi_open_escapable_handle_scope [/.nvm/versions/node/v16.20.0/bin/node]
 6: 0x39b1ac2c9 Napi::FunctionReference::New(std::initializer_list<napi_value__*> const&) const [/node_modules/@datadog/native-appsec/prebuilds/darwin-x64/node-napi.node]
 7: 0x39b1ac1f0 DDWAF::createContext(Napi::CallbackInfo const&) [/node_modules/@datadog/native-appsec/prebuilds/darwin-x64/node-napi.node]
 8: 0x39b1adb6f napi_value__* Napi::details::TemplatedInstanceCallback<DDWAF, &(DDWAF::createContext(Napi::CallbackInfo const&))>(napi_env__*, napi_callback_info__*) [/node_modules/@datadog/native-appsec/prebuilds/darwin-x64/node-napi.node]
 9: 0x10e0cc050 v8impl::(anonymous namespace)::FunctionCallbackWrapper::Invoke(v8::FunctionCallbackInfo<v8::Value> const&) [/.nvm/versions/node/v16.20.0/bin/node]
10: 0x10e2edc29 v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) [/.nvm/versions/node/v16.20.0/bin/node]
11: 0x10e2ed6f6 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [/.nvm/versions/node/v16.20.0/bin/node]
12: 0x10e2ece6f v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [/.nvm/versions/node/v16.20.0/bin/node]
13: 0x10eb5f579 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit [/.nvm/versions/node/v16.20.0/bin/node]
14: 0x10eaf1f6a Builtins_InterpreterEntryTrampoline [/.nvm/versions/node/v16.20.0/bin/node]
15: 0x10eaf1f6a Builtins_InterpreterEntryTrampoline [/.nvm/versions/node/v16.20.0/bin/node]
16: 0x10eaf1f6a Builtins_InterpreterEntryTrampoline [/.nvm/versions/node/v16.20.0/bin/node]
```

**Steps to reproduce**

This particular app is using worker threads, not sure if that is related. When appsec mode is enabled the above exception is thrown.

**Environment**

* **Operation system:** Mac OS
* **Node.js version:** v16.20.0
* **Tracer version:** 4.2.0
* **Agent version:** N/A running locally
* **Relevant library versions:**

Same as https://github.com/DataDog/dd-native-appsec-js/issues/82

**Comment by simon-id** [comment]
Thank you for reporting this bug. Indeed the usage of `worker_threads` was triggering an edge case in our code. This has been fixed and will be released in the next version of `dd-trace`.

**Comment by simon-id** [comment]
The fix has been released in version `5.1.0`/`4.25.0`/`3.46.0` of dd-trace! Please reopen the issue if you're still having problems.
